### ISSUE TITLE
Bastion object for foreign shoot in same project blocks shoot deletion

### ISSUE THREAD

**Issue by MartinWeindel** [gardener/gardener#5353]
**How to categorize this issue?**

/area control-plane
/kind bug

**What happened**:
I wanted to delete a shoot cluster, but the deletion was blocked with an error in the gardenlet log
```txt
{"level":"info","msg":"Error syncing Shoot garden-core/e5t7y1878n: shoot has still Bastions","ts":"2022-01-31T10:24:41.815Z"}
```
There is no bastion object for this shoot, but only for another one:
```txt
# on virtual garden
$ k -n garden-core get bastion
NAME           SHOOT          SEED   IP          HEARTBEAT   EXPIRES     AGE
cli-qmc0yqet   d062553-aws1   aws    <pending>   6d17h ago   <expired>   6d17h
```

**What you expected to happen**:
The shoot cluster deletion is not blocked by a bastion of a foreign shoot in the same project namespace.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
This seems to be a bug in the method `shootReconciler.shootHasBastions`:
https://github.com/gardener/gardener/blob/66b4624eab3deb166aca90db460b4beef901c68e/pkg/gardenlet/controller/shoot/shoot_control.go#L455
The `shoot.namespace` is the namespace of the garden project. There is a restriction missing for the shoot itself.

**Environment**:

- Gardener version: v1.39.1

**Comment by ialidzhikov** [comment]
Duplicate of https://github.com/gardener/gardener/issues/5323.
/close
